### ISSUE TITLE
How to define a string input to the exported onnx model?

### ISSUE THREAD

**Issue by Monah84** [microsoft/onnxruntime#5489]
**Describe the bug**
I am facing a problem while exporting the BERT model; I want to define a string input so I can pass it through the InferenceSession from ASP.NET Core. When I pass the string to the ```session.Run```, I get an error 

> Microsoft.ML.OnnxRuntime.OnnxRuntimeException
>   HResult=0x80131500
>   Message=[ErrorCode:InvalidArgument] Unexpected input data type. Actual: (class onnxruntime::PrimitiveDataType<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >) , expected: (class onnxruntime::PrimitiveDataType<__int64>)
>   Source=Microsoft.ML.OnnxRuntime
>   StackTrace:
>    at Microsoft.ML.OnnxRuntime.NativeApiStatus.VerifySuccess(IntPtr nativeStatus)
>    at Microsoft.ML.OnnxRuntime.InferenceSession.RunImpl(RunOptions options, IntPtr[] inputNames, IntPtr[] inputValues, IntPtr[] outputNames, DisposableList`1 cleanupList)
>    at Microsoft.ML.OnnxRuntime.InferenceSession.Run(IReadOnlyCollection`1 inputs, IReadOnlyCollection`1 outputNames, RunOptions options)
>    at Microsoft.ML.OnnxRuntime.InferenceSession.Run(IReadOnlyCollection`1 inputs, IReadOnlyCollection`1 outputNames)
>    at Microsoft.ML.OnnxRuntime.InferenceSession.Run(IReadOnlyCollection`1 inputs)

**To Reproduce**
```
output_model_path = "model/model.onnx"

model.model.eval()

inputs = {
        'input_ids':   torch.randint(32, [2, 32], dtype=torch.long).to(device), <--- HERE MY PROBLEM
        'attention_mask': torch.ones([2, 32], dtype=torch.long).to(device),  
        'token_type_ids':  torch.ones([2, 32], dtype=torch.long).to(device),  
    }

torch.onnx.export(model.model,     
                  (inputs['input_ids'], 
                   inputs['attention_mask'], 
                   inputs['token_type_ids']),     
                  output_model_path,           
                  opset_version=11,          
                  do_constant_folding=True,       
                  input_names=['input'],           
                  output_names=['predictions','raw_output'])        
```
**Expected behavior**
To define an input with data type dtype=torch.str for example?

**Comment by BowenBao** [comment]
hi @Monah84, based on your description it seems the issue is more suited under the huggingface repo, since your ask is to extend the original model inputs dtype with torch.str. 
That being said, are you looking for str type because the input you have is raw sentences in string? If so you should use the tokenizer to preprocess the string into token ids.

**Comment by Monah84** [comment]
Yes, you are right. I confused me since during calling the predict function, It accepts raw string. But, when I converted it to onnx, it does not. I converted the raw string to tokens using the tokenizer. Now, I am trying to figure out why the predict function returns array of float "probability" and not array of 0 and 1 for the multi-label classification.

Thank you for the reply, it was very helpful
